### ISSUE TITLE
How to avoid "degenerate solution on iteration 2, probably too small l2-regularization, try to increase it"

### ISSUE THREAD

**Issue by unnir** [catboost/catboost#193]
Hi all, 

I'm facing this error even with a high value of l2_leaf_reg from 30 to 100. 
Any recommendation how to avoid it?

**Comment by kizill** [comment]
Hello, @unnir. What loss function do you use?

**Comment by unnir** [comment]
@kizill  Hi, 

here is the output from model.get_params()

```
{'depth': 3,
 'feature_border_type': 'MinEntropy',
 'has_time': False,
 'iterations': 3,
 'l2_leaf_reg': 64,
 'learning_rate': 2.5,
 'logging_level': 'Silent',
 'loss_function': 'Logloss',
 'name': 'experiment',
 'random_seed': 1111,
 'random_strength': 1}
```

**Comment by kizill** [comment]
I strongly recommend you to use learning rates less then 1.0

**Comment by unnir** [comment]
yep, it helped. 

Thank you for your advice.
